### PR TITLE
Update examples

### PR DESCRIPTION
**Pull request by benlesh**
Updates the examples to reflect recent changes. Also ensures that we have imperative versions for each example.

Adds a new example that creates a measuring tool to measure distance on the page.

### CODE REVIEW

**Comment by domfarolino** [comment]
It looks like there are conflicts in this PR now because this PR somehow contained all of the "prettier" changes that the PR did.

**Comment by domfarolino** [comment]
Ping :)

**Comment by benlesh** [comment]
Weird... That diff was gnarly. I just redid the changes on top of master.

File: README.md [hunk 1]
```diff
@@ -273,6 +322,120 @@ document.addEventListener('keydown', e => {
...
+When you mousedown on the document, it will start measuring how far your mouse moves, rendering a
+line and some text for how far you've measured, and when you mouse up, will log the final
+distance before removing the line.
+
+```js
+const measurements = document.on('mousedown').flatMap((e) => {
+	const { clientX: startX, clientY: startY } = e;
+
+	const svg = document.createElement('svg');
+	svg.width = document.body.width;
```
[the remark this answers is not in the document]
Comment (benlesh) [reply README.md:336]: No... I'm making things up, clearly. 😅

File: README.md [hunk 2]
```diff
@@ -23,62 +23,88 @@ hard-to-follow callback chains.
...
+  }));
+```
+
+<details>
+<summary>Imperative version</summary>
+
+```js
+element.addEventListener('click', (e) => {
+	if (e.target.matches('.foo')) {
+		const { clientX: x, clientY: y } = e;
```
[the remark this answers is not in the document]
Comment (benlesh) [reply README.md:40]: This doesn't work :)

File: README.md [hunk 3]
```diff
@@ -23,62 +23,88 @@ hard-to-follow callback chains.
...
 ```js
 // Filtering and mapping:
-element
-	.on('click')
-	.filter((e) => e.target.matches('.foo'))
-	.map((e) => ({ x: e.clientX, y: e.clientY }))
-	.subscribe({ next: handleClickAtPoint });
+element.on('click')
+  .filter(e => e.target.matches('.foo'))
+  .map(e => ({x: e.clientX, y: e.clientY }))
```
[the remark this answers is not in the document]
Comment (benlesh) [reply README.md:28]: We can change it... this is just the default for prettier.

Basically, it's goal is to reduce noise if possible. So it only adds `()` when it needs to: More than one argument, some destructuring, types included, etc.

[the remark this answers is not in the document]
Comment (domfarolino) [reply README.md:28]: > So it only adds `()` when it needs to: More than one argument,

That is inconsistent with most of the changes in the original prettier PR, for example https://github.com/WICG/observable/pull/63/files#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R56 and many more there.

File: README.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (domfarolino) [reply README.md:40]: Ah it's so weird that `e` has an `x` and `clientX` property, which confuses me.

File: README.md [hunk 4]
```diff
@@ -23,62 +23,88 @@ hard-to-follow callback chains.
...
 ```js
-// Imperative
-const controller = new AbortController();
+const ac = new AbortController();
 element.addEventListener(
 	'mousemove',
 	(e) => {
-		element.addEventListener('mouseup', (e) => controller.abort());
+		element.addEventListener('mouseup', (e) => ac.abort(), {
+			signal: controller.signal,
```
[the remark this answers is not in the document]
Comment (domfarolino) [reply README.md]: Looks like you didn't get to this:

> But also, why do we have to pass it in here anyways, if the previous version of this example wasn't?

File: README.md [hunk 5]
```diff
@@ -273,6 +322,120 @@ document.addEventListener('keydown', e => {
...
+
+```js
+const ac = new AbortController();
+document.addEventListener(
+	'mousedown',
+	(e) => {
+		const { clientX: startX, clientY: startY } = e;
+
+		const svg = document.createElement('svg');
+		svg.width = document.body.width;
```
Comment (domfarolino) [inline README.md:391]: See https://github.com/WICG/observable/pull/83#discussion_r1412849609

**Comment by domfarolino** [comment]
@benlesh do you plan on finishing this?

**Comment by domfarolino** [comment]
I'm going to close this since there is no activity. Feel free to re-open if you want to finish this.

**Comment by benlesh** [comment]
Sorry... I've just been busy with personal life things. I'll try to circle back to this soon.

**Pull request closed without merging**